addons/mass_mailing/models/mailing_list.py

```diff
@@ -219,10 +240,44 @@ def action_merge(self, src_lists, archive):
                     )
                 ) st
             WHERE st.rn = 1;""", (self.id, tuple(src_lists.ids), self.id))
+        self.env.cr.execute("""
+            UPDATE mailing_contact_list_rel
```

Comment:
Same remark as other update in sms, would be cool to do everything in a single query